signer-service/src/api/services/sep10.service.js

```diff
@@ -0,0 +1,95 @@
+const { Keypair } = require('stellar-sdk');
+const { TransactionBuilder, Networks } = require('stellar-sdk');
+const { fetchTomlValues } = require('../helpers/anchors');
+
+const { TOKEN_CONFIG } = require('../../constants/tokenConfig');
+const { CLIENT_SECRET } = require('../../constants/constants');
+
+const NETWORK_PASSPHRASE = Networks.PUBLIC;
+
+exports.signSep10Challenge = async (challengeXDR, outToken, clientPublicKey) => {
+  const clientDomainStellarKeypair = Keypair.fromSecret(CLIENT_SECRET);
+
+  const { signingKey: anchorSigningKey } = await fetchTomlValues(TOKEN_CONFIG[outToken].tomlFileUrl);
+
+  const transactionSigned = new TransactionBuilder.fromXDR(challengeXDR, NETWORK_PASSPHRASE);
+  if (transactionSigned.source !== anchorSigningKey) {
+    throw new Error(`Invalid source account: ${transactionSigned.source}`);
+  }
+  if (transactionSigned.sequence !== '0') {
+    throw new Error(`Invalid sequence number: ${transactionSigned.sequence}`);
+  }
+
+  // See https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md#success
+  // memo field should be empty as we assume (in this implementation) that we use the ephemeral
+  // to authenticate. But no memo sub account derivation.
+  if (transactionSigned.memo.value === '') {
+    throw new Error('Memo does not match');
+  }
+
+  const { operations } = transactionSigned;
+  // Verify the first manage_data operation
+  const firstOp = operations[0];
+  if (firstOp.type !== 'manageData') {
+    throw new Error('The first operation should be manageData');
+  }
+
+  // Only authorize a session that corresponds with the ephemeral client account
+  if (firstOp.source !== clientPublicKey) {
+    throw new Error('First manageData operation must have the client account as the source');
+  }
+
+  const { anchorExpectedKey: expectedKey, clientDomainEnabled } = TOKEN_CONFIG[outToken];
+  if (firstOp.name !== expectedKey) {
```

Comment:
Are you sure it's called `name` and not `key`? The protocol [says](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md#success) it's supposed to be `key` but maybe it's different in practice.

```suggestion
  if (firstOp.key !== expectedKey) {
```

Reply:
Yup, name. Here is an example

```
{
    source: 'GA6KB5YJ6B2CBJK7TZT2UWT7TWS4PNJVWHDMLVQ42ESMDD55XWJXJUWR',
    type: 'manageData',
    name: 'api.anclap.com auth',
    value: <...... more bytes>
  },
  ```